velox/common/memory/MemoryPool.cpp

```diff
@@ -809,15 +809,9 @@ bool MemoryPoolImpl::incrementReservationThreadSafe(
     TestValue::adjust(
         "facebook::velox::memory::MemoryPoolImpl::incrementReservationThreadSafe::AfterGrowCallback",
         this);
-    // NOTE: the memory reservation might still fail even if the memory grow
-    // callback succeeds. The reason is that we don't hold the root tracker's
-    // mutex lock while running the grow callback. Therefore, there is a
-    // possibility in theory that a concurrent memory reservation request
-    // might steal away the increased memory capacity after the grow callback
-    // finishes and before we increase the reservation. If it happens, we can
-    // simply fall back to retry the memory reservation from the leaf memory
-    // pool which should happen rarely.
-    return maybeIncrementReservation(size);
```

Comment:
The change caused regression in Gluten: https://github.com/apache/incubator-gluten/pull/5558. We are working on fixing.